scout/commands/delete_case.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+# encoding: utf-8
+"""
+delete_case.py
+
+Script to delete a case and all of its variants from the database.
+
+Created by Måns Magnusson on 2015-04-07.
+Copyright (c) 2015 ScoutTeam__. All rights reserved.
+
+"""
+
+
+from __future__ import (absolute_import, unicode_literals, print_function)
+
+import sys
+import os
+import logging
+
+import click
+
+from pymongo import MongoClient, Connection
+from pymongo.errors import CollectionInvalid,PyMongoError
+from mongoengine import connect, DoesNotExist
+from mongoengine.connection import _get_db
+
+from ..models import (Case, Variant)
+
+def remove_case(case_id, owner, mongo_db='variantDatabase', username=None, 
+                password=None, port=27017, host='localhost', logger=None):
+  """Delete variants and users from the mongo database."""
+  # get root path of the Flask app
+  # project_root = '/'.join(app.root_path.split('/')[0:-1])
+  if not logger:
+    logger = logging.getLogger(__name__)
+  
+  logger.info('Trying to access collection {0}'.format(mongo_db))
+
+  connection = connect(
+                    mongo_db, 
+                    host=host, 
+                    port=port, 
+                    username=username,
+                    password=password
+                    )
+
+  logger.debug('Connection successful')
+  
+  case_mongo_id = '_'.join([owner, case_id])
+  
+  databases = connection.database_names()
+  logger.info("Databases found: {0}".format(', '.join(databases)))
+  if mongo_db in databases:
+    logger.info("Accsessing {0}".format(mongo_db))
+    db = connection[mongo_db]
+    logger.debug("Access succesful")
+    
+    logger.info("Trying to find case {0} in database".format(case_mongo_id))
+    case = Case.objects(case_id=case_mongo_id)
+    if case:
+      logger.info("Case found! Deleting case")
+      case.delete()
+      logger.debug("Case deleted")
+    else:
+      logger.info("Case {0} not found in database".format(case_mongo_id))
+    
+    nr_of_variants = 0
+    for variant in Variant.objects(case_id = case_mongo_id):
```

Comment:
I guess it would not be possible to log the deletions but please change if you like